ocrd_webapi/authentication.py

```diff
@@ -0,0 +1,55 @@
+from hashlib import sha512
+from random import random
+from typing import Tuple
+
+from .database import create_user, get_user
+
+
+async def authenticate_user(email: str, password: str):
+    db_user = await get_user(email=email)
+    if not db_user:
+        raise ValueError(f"User not found: {email}")
```

Comment:
I would suggest raising a custom error for authentication related problems, e.g. `AuthenticationError`. Same for registration (`RegistrationError`).

Reply:
Good point.